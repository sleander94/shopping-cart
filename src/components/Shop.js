import React, { useState } from 'react';
import Product from './Product';
import productList from '../products';
import '../styles/Shop.css';

const Shop = () => {
  const [cartProducts, setCartProducts] = useState([]);

  const addToCart = (index) => {
    let added = false;
    const product = productList[index];
    cartProducts.forEach((item) => {
      if (product.name === item.name) {
        item.quantity += 1;
        added = true;
      }
    });

    if (added === false) {
      product.quantity = 1;
      setCartProducts([...cartProducts, product]);
    }
    console.log(cartProducts);
  };
  return (
    <div id="shop">
      {productList.map((product) => (
        <Product
          key={productList.indexOf(product)}
          {...product}
          index={productList.indexOf(product)}
          addToCart={addToCart}
        />
      ))}
    </div>
  );
};

export default Shop;
