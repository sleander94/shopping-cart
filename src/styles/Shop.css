#shop {
  display: grid;
  justify-items: center;
  background: rgb(240, 240, 240);
  height: 100%;
}

.products {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 34px;
  padding-top: 60px;
  padding-bottom: 60px;
  min-width: 640px;
  max-width: 1400px;
  background: rgb(240, 240, 240);
}
