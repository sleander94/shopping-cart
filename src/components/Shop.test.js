import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Shop from './Shop';

describe('Shop component', () => {
  it('renders correctly', () => {
    const shop = renderer.create(<Shop />);
    expect(shop).toMatchSnapshot();
  });
});
