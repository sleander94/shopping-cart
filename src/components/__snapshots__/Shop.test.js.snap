// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders correctly 1`] = `
<div
  id="shop"
>
  <div
    className="products"
  >
    <div
      className="product"
    >
      <img
        alt="green tent"
        src="4tent.jpeg"
      />
      <div
        className="name"
      >
        Four Person Camping Tent
      </div>
      <div
        className="price"
      >
        $
        349.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="orange and blue tent"
        src="6tent.jpeg"
      />
      <div
        className="name"
      >
        Six Person Luxury Tent
      </div>
      <div
        className="price"
      >
        $
        499.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="yellow and black sleeping bag"
        src="sleepingbag.jpeg"
      />
      <div
        className="name"
      >
        Sub-Zero Sleeping Bag
      </div>
      <div
        className="price"
      >
        $
        99.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="orange and blue double sleeping bag"
        src="doublebag.jpeg"
      />
      <div
        className="name"
      >
        Double Sleeping Bag
      </div>
      <div
        className="price"
      >
        $
        165.49
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="beige trekking backpack"
        src="mbackpack.jpeg"
      />
      <div
        className="name"
      >
        Mens Trekking Pack 60L
      </div>
      <div
        className="price"
      >
        $
        219.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="teal trekking backpack"
        src="wbackpack.jpeg"
      />
      <div
        className="name"
      >
        Womens Trekking Pack 60L
      </div>
      <div
        className="price"
      >
        $
        229.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="extendable trekking poles with wrist straps"
        src="poles.jpeg"
      />
      <div
        className="name"
      >
        Carbon Fiber Trekking Poles
      </div>
      <div
        className="price"
      >
        $
        199.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt=" red jacket"
        src="jacket.jpeg"
      />
      <div
        className="name"
      >
        High Visibility Puffer Jacket
      </div>
      <div
        className="price"
      >
        $
        79.95
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="double burner red camp stove"
        src="stove.jpeg"
      />
      <div
        className="name"
      >
        Portable Gas Camp Stove
      </div>
      <div
        className="price"
      >
        $
        89.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="black binoculars"
        src="binoculars.jpeg"
      />
      <div
        className="name"
      >
        Long Range Binoculars
      </div>
      <div
        className="price"
      >
        $
        219.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="gps displaying color map"
        src="gps.jpeg"
      />
      <div
        className="name"
      >
        Weather Proof Handhelf GPS
      </div>
      <div
        className="price"
      >
        $
        699.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
    <div
      className="product"
    >
      <img
        alt="multitool displaying pliers and knives"
        src="multitool.jpeg"
      />
      <div
        className="name"
      >
        Stainless Steel Multitool
      </div>
      <div
        className="price"
      >
        $
        149.99
      </div>
      <button
        className="addToCart"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
